lib/xandra/connection/utils.ex

```diff
@@ -3,9 +3,11 @@ defmodule Xandra.Connection.Utils do
 
   alias Xandra.{Connection.Error, Frame, Protocol}
 
-  @spec recv_frame(:gen_tcp.socket) ::
+  @spec recv_frame(:gen_tcp.socket, nil | module) ::
         {:ok, Frame.t} | {:error, :closed | :inet.posix}
-  def recv_frame(socket) do
+  def recv_frame(socket, compressor \\ nil)
+      when is_nil(compressor)
```

Comment:
I personally don't like the idea of this, because it's clearer like this than having to figure out that `is_atom/1` works because `nil` is an atom, but I guess if it raised questions it's not that clear after all.